libcloudforensics/providers/azure/internal/account.py

```diff
@@ -37,16 +44,44 @@ class AZAccount:
     compute_client (ComputeManagementClient): An Azure compute client object.
   """
 
-  def __init__(self, subscription_id: str) -> None:
+  def __init__(self,
+               default_resource_group_name: str,
+               default_region: Optional[str] = None,
+               profile_name: Optional[str] = None) -> None:
     """Initialize the AZAccount class.
 
     Args:
-      subscription_id (str): The Azure subscription ID to use.
+      default_resource_group_name (str): The default resource group in which to
+          create new resources in. If the resource group does not exists,
+          it will be automatically created.
+      default_region (str): Optional. The default region to create new
+          resources in. If None, it will default to eastus.
+      profile_name (str): Optional. The name of the profile to use for Azure
+          operations. For more information on profiles, see GetCredentials()
+          in libcloudforensics.providers.azure.internal.common.py. Default
+          does not use profiles and will authenticate to Azure using
+          environment variables.
     """
-    self.subscription_id = subscription_id
-    self.credentials = common.GetCredentials()
-    self.compute_client = ComputeManagementClient(
+    self.subscription_id, self.credentials = common.GetCredentials(profile_name)
+    self.default_region = default_region or 'eastus'
```

Comment:
the 'eastus' default is buried pretty deep here, is there no way to make that clearer in the calling signature of the function on some level?

Reply:
Updated as default parameter value for more visibility